src/main/java/bacnet/EditablePoint.java

```diff
@@ -156,25 +176,37 @@ public void handle(ActionResult event) {
 			makeCopy();
 		}
 	}
-	
+
 	protected class EditHandler implements Handler<ActionResult> {
 		public void handle(ActionResult event) {
-			String newname = event.getParameter("name", ValueType.STRING).getString();
-			if (newname != null && newname.length() > 0 && !newname.equals(node.getName())) {
-				parent.removeChild(node);
-				node = parent.createChild(newname).build();
+			String name = event.getParameter(ATTRIBUTE_NAME, ValueType.STRING).getString();
+			if (name != null && name.length() > 0 && !name.equals(node.getName())) {
+				duplicate(name);
 			}
-			
+			objectTypeDescription = Utils.parseObjectType(event.getParameter(ATTRIBUTE_OBJECT_TYPE).getString())
+					.toString();
+			instanceNumber = event.getParameter(ATTRIBUTE_OBJECT_INSTANCE_NUMBER, ValueType.NUMBER).getNumber()
+					.intValue();
+			cov = event.getParameter(ATTRIBUTE_USE_COV, ValueType.BOOL).getBool();
+			settable = event.getParameter(ATTRIBUTE_SETTABLE, ValueType.BOOL).getBool();
+
 			setupNode();
+		}
+	}
 
+	private void duplicate(String newname) {
+		if (newname != null && newname.length() > 0 && !newname.equals(node.getName())) {
```

Comment:
I feel like this if condition is repeated over and over. Maybe wrap it in a utility method would make the code much cleaner.